src/mini_s3.erl

```diff
@@ -115,327 +131,153 @@ manual_start() ->
     application:start(ssl),
     application:start(inets).
 
--spec new(string(), string()) -> config().
-
-new(AccessKeyID, SecretAccessKey) ->
-    #config{
-     access_key_id=AccessKeyID,
-     secret_access_key=SecretAccessKey}.
-
--spec new(string(), string(), string()) -> config().
-
-new(AccessKeyID, SecretAccessKey, Host) ->
-    #config{
-     access_key_id=AccessKeyID,
-     secret_access_key=SecretAccessKey,
-     s3_url=Host}.
-
--spec new(string(), string(), string(), bucket_access_type()) -> config().
+-spec new(string() | binary(), string() | binary(), string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host0) ->
+    % chef-server crams scheme://host:port all into into Host; erlcloud wants them separate.
+    % Assume:
+    %   Host   == scheme://domain:port | scheme://domain | domain:port | domain
+    %   scheme == http | https
+
+    % ipv4/6 detection
+    {Ipv, Host} =
```

Comment:
It might be nice to pull this URI parsing code out into its own function.

Reply:
Done.
https://buildkite.com/chef-oss/chef-mini-s3-master-verify/builds/9
https://buildkite.com/chef/chef-chef-server-master-verify/builds/2437#2f07bc74-f096-43b5-8848-33dcbf628443
https://buildkite.com/chef/chef-chef-server-master-omnibus-adhoc/builds/1667#7b94b4e5-b553-4482-b187-09fbd6fc3a5c